### PR TITLE
snapshot: deprecate forward compatibility

### PR DESCRIPTION
**Pull request by bchalios**
## Changes

Forward compatibility refers to the ability of Firecracker to create snapshots for older version formats. This feature is now deprecated and it will be removed in a subsequent release. Snapshot support is in developer preview, so dropping support for it does not require us to increase Firecracker's major version.

In practice, deprecating this feature means that we deprecate the `version` field of the `PUT /snapshot/create` request. This commit, adds the "deprecated" header in the response of the endpoint if `version` is defined and documents the deprecation in CHANGELOG and snapshot documentation.

## License Acceptance

By submitting this pull request, I confirm that my contribution is made under
the terms of the Apache 2.0 license. For more information on following
Developer Certificate of Origin and signing off your commits, please check
[`CONTRIBUTING.md`][3].

## PR Checklist

- [ ] If a specific issue led to this PR, this PR closes the issue.
- [ ] The description of changes is clear and encompassing.
- [ ] Any required documentation changes (code and docs) are included in this PR.
- [ ] API changes follow the [Runbook for Firecracker API changes][2].
- [ ] User-facing changes are mentioned in `CHANGELOG.md`.
- [ ] All added/changed functionality is tested.
- [ ] New `TODO`s link to an issue.
- [ ] Commits meet [contribution quality standards](https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md#contribution-quality-standards).

---

- [ ] This functionality cannot be added in [`rust-vmm`][1].

[1]: https://github.com/rust-vmm
[2]: https://github.com/firecracker-microvm/firecracker/blob/main/docs/api-change-runbook.md
[3]: https://github.com/firecracker-microvm/firecracker/blob/main/CONTRIBUTING.md

### CODE REVIEW

**Review by kalyazin — DISMISSED** [review]

**Review by ShadowCurse — COMMENTED** [review]
Shouldn't there be also changes in `vmm/src/persist.rs`? At least `extra_version_check` ant `get_snapshot_data_version` can be removed.

**Comment by ShadowCurse** [comment]
Oh yeah, my mistake. We should remove it in the next release, not this one.

File: src/api_server/swagger/firecracker.yaml [hunk 1]
```diff
@@ -1179,7 +1179,8 @@ definitions:
         type: string
         description:
           The microVM version for which we want to create the snapshot.
-          It is optional and it defaults to the current version.
+          It is optional and it defaults to the current version. This parameter
+          has been deprecated and it will be removed in future Firecracker release.
```
Comment (ShadowCurse) [inline src/api_server/swagger/firecracker.yaml:1183]: maybe specify the firecracker version when it was deprecated?

Comment (bchalios) [reply src/api_server/swagger/firecracker.yaml:1183]: I thought about it but (1) other similar messages don't mention it and (2) after thinking about it, I don't think it's really useful information. But I don't have strong opinions

File: src/api_server/src/request/snapshot.rs [hunk 2]
```diff
@@ -14,6 +14,9 @@ use crate::request::{Body, Method, StatusCode};
 
 /// Deprecation message for the `mem_file_path` field.
 const LOAD_DEPRECATION_MESSAGE: &str = "PUT /snapshot/load: mem_file_path field is deprecated.";
+/// Deprecation message for the `version` field.
+const CREATE_WITH_VERSION_DEPRECATION_MESSAGE: &str =
+    "PUT /snapshot/create: 'version' field is deprecated.";
```
[the remark this answers is not in the document]
Comment (bchalios) [reply src/api_server/src/request/snapshot.rs:19]: same as before. WDYT?

**Review by roypat — APPROVED** [review]

File: src/api_server/swagger/firecracker.yaml (hunk 1 above)
Comment (ShadowCurse) [reply src/api_server/swagger/firecracker.yaml:1183]: Ok, I don't have a strong preference either.

**Review by kalyazin — APPROVED** [review]

**Pull request merged**